app/models/donation.rb

```diff
@@ -43,7 +40,7 @@ def process_payment
   end
 
   def record_donation
-    person = { email: email, employer: employer, occupation: occupation }
+    person = { email: self.person.email, employer: employer, occupation: occupation }
```

Comment:
could rename `person` local variable and not need to call self in `self.person.email`